### ISSUE TITLE
PostgreSQL 16 support

### ISSUE THREAD

**Issue by devrimgunduz** [pgexporter/pgexporter_ext#11]
Hi,

pgexporter_ext fails to build against PostgreSQL 16, and v16 is due next week. Can you please take a look?

Thanks!

Regards, Devrim

> [ 66%] Building C object src/CMakeFiles/pgexporter_ext.dir/pgexporter_ext/utils.c.o
[ 66%] Building C object src/CMakeFiles/pgexporter_ext.dir/pgexporter_ext/lib.c.o
In file included from rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:47:
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c: In function 'pgexporter_information_ext':
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:175:22: warning: passing argument 1 of 'CStringGetDatum' makes pointer from integer without a cast [-Wint-conversion]
  175 |    PG_RETURN_CSTRING(version);
      |                      ^~~~~~~
      |                      |
      |                      Datum {aka long unsigned int}
/usr/pgsql-16/include/server/fmgr.h:362:53: note: in definition of macro 'PG_RETURN_CSTRING'
  362 | #define PG_RETURN_CSTRING(x) return CStringGetDatum(x)
      |                                                     ^
In file included from rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:46:
/usr/pgsql-16/include/server/postgres.h:350:29: note: expected 'const char *' but argument is of type 'Datum' {aka 'long unsigned int'}
  350 | CStringGetDatum(const char *X)
      |                 ~~~~~~~~~~~~^
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c: In function 'pgexporter_version_ext':
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:189:22: warning: passing argument 1 of 'CStringGetDatum' makes pointer from integer without a cast [-Wint-conversion]
  189 |    PG_RETURN_CSTRING(version);
      |                      ^~~~~~~
      |                      |
      |                      Datum {aka long unsigned int}
/usr/pgsql-16/include/server/fmgr.h:362:53: note: in definition of macro 'PG_RETURN_CSTRING'
  362 | #define PG_RETURN_CSTRING(x) return CStringGetDatum(x)
      |                                                     ^
/usr/pgsql-16/include/server/postgres.h:350:29: note: expected 'const char *' but argument is of type 'Datum' {aka 'long unsigned int'}
  350 | CStringGetDatum(const char *X)
      |                 ~~~~~~~~~~~~^
In file included from /usr/pgsql-16/include/server/postgres.h:45:
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c: In function 'cpu_info':
/usr/pgsql-16/include/server/c.h:916:14: error: static assertion failed: "cpu_freq does not have type int64"
  916 |         do { _Static_assert(condition, errmessage); } while(0)
      |              ^~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:918:20: note: in expansion of macro 'StaticAssertStmt'
  918 |         ((void) ({ StaticAssertStmt(condition, errmessage); true; }))
      |                    ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:962:10: note: in expansion of macro 'StaticAssertExpr'
  962 |         (StaticAssertExpr(__builtin_types_compatible_p(__typeof__(varname), typename), \
      |          ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/postgres.h:550:10: note: in expansion of macro 'AssertVariableIsOfTypeMacro'
  550 |         (AssertVariableIsOfTypeMacro(X, int64), Int64GetDatum(X))
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:671:38: note: in expansion of macro 'Int64GetDatumFast'
  671 |       values[CPU_INFO_CLOCK_SPEED] = Int64GetDatumFast(cpu_freq);
      |                                      ^~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c: In function 'memory_info':
/usr/pgsql-16/include/server/c.h:916:14: error: static assertion failed: "total_memory_bytes does not have type int64"
  916 |         do { _Static_assert(condition, errmessage); } while(0)
      |              ^~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:918:20: note: in expansion of macro 'StaticAssertStmt'
  918 |         ((void) ({ StaticAssertStmt(condition, errmessage); true; }))
      |                    ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:962:10: note: in expansion of macro 'StaticAssertExpr'
  962 |         (StaticAssertExpr(__builtin_types_compatible_p(__typeof__(varname), typename), \
      |          ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/postgres.h:550:10: note: in expansion of macro 'AssertVariableIsOfTypeMacro'
  550 |         (AssertVariableIsOfTypeMacro(X, int64), Int64GetDatum(X))
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:820:42: note: in expansion of macro 'Int64GetDatumFast'
  820 |       values[MEMORY_INFO_TOTAL_MEMORY] = Int64GetDatumFast(total_memory_bytes);
      |                                          ^~~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:916:14: error: static assertion failed: "used_memory_bytes does not have type int64"
  916 |         do { _Static_assert(condition, errmessage); } while(0)
      |              ^~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:918:20: note: in expansion of macro 'StaticAssertStmt'
  918 |         ((void) ({ StaticAssertStmt(condition, errmessage); true; }))
      |                    ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:962:10: note: in expansion of macro 'StaticAssertExpr'
  962 |         (StaticAssertExpr(__builtin_types_compatible_p(__typeof__(varname), typename), \
      |          ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/postgres.h:550:10: note: in expansion of macro 'AssertVariableIsOfTypeMacro'
  550 |         (AssertVariableIsOfTypeMacro(X, int64), Int64GetDatum(X))
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:821:41: note: in expansion of macro 'Int64GetDatumFast'
  821 |       values[MEMORY_INFO_USED_MEMORY] = Int64GetDatumFast(used_memory_bytes);
      |                                         ^~~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:916:14: error: static assertion failed: "free_memory_bytes does not have type int64"
  916 |         do { _Static_assert(condition, errmessage); } while(0)
      |              ^~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:918:20: note: in expansion of macro 'StaticAssertStmt'
  918 |         ((void) ({ StaticAssertStmt(condition, errmessage); true; }))
      |                    ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:962:10: note: in expansion of macro 'StaticAssertExpr'
  962 |         (StaticAssertExpr(__builtin_types_compatible_p(__typeof__(varname), typename), \
      |          ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/postgres.h:550:10: note: in expansion of macro 'AssertVariableIsOfTypeMacro'
  550 |         (AssertVariableIsOfTypeMacro(X, int64), Int64GetDatum(X))
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:822:41: note: in expansion of macro 'Int64GetDatumFast'
  822 |       values[MEMORY_INFO_FREE_MEMORY] = Int64GetDatumFast(free_memory_bytes);
      |                                         ^~~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:916:14: error: static assertion failed: "swap_total_bytes does not have type int64"
  916 |         do { _Static_assert(condition, errmessage); } while(0)
      |              ^~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:918:20: note: in expansion of macro 'StaticAssertStmt'
  918 |         ((void) ({ StaticAssertStmt(condition, errmessage); true; }))
      |                    ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:962:10: note: in expansion of macro 'StaticAssertExpr'
  962 |         (StaticAssertExpr(__builtin_types_compatible_p(__typeof__(varname), typename), \
      |          ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/postgres.h:550:10: note: in expansion of macro 'AssertVariableIsOfTypeMacro'
  550 |         (AssertVariableIsOfTypeMacro(X, int64), Int64GetDatum(X))
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:823:40: note: in expansion of macro 'Int64GetDatumFast'
  823 |       values[MEMORY_INFO_SWAP_TOTAL] = Int64GetDatumFast(swap_total_bytes);
      |                                        ^~~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:916:14: error: static assertion failed: "swap_used_bytes does not have type int64"
  916 |         do { _Static_assert(condition, errmessage); } while(0)
      |              ^~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:918:20: note: in expansion of macro 'StaticAssertStmt'
  918 |         ((void) ({ StaticAssertStmt(condition, errmessage); true; }))
      |                    ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:962:10: note: in expansion of macro 'StaticAssertExpr'
  962 |         (StaticAssertExpr(__builtin_types_compatible_p(__typeof__(varname), typename), \
      |          ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/postgres.h:550:10: note: in expansion of macro 'AssertVariableIsOfTypeMacro'
  550 |         (AssertVariableIsOfTypeMacro(X, int64), Int64GetDatum(X))
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:824:39: note: in expansion of macro 'Int64GetDatumFast'
  824 |       values[MEMORY_INFO_SWAP_USED] = Int64GetDatumFast(swap_used_bytes);
      |                                       ^~~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:916:14: error: static assertion failed: "swap_free_bytes does not have type int64"
  916 |         do { _Static_assert(condition, errmessage); } while(0)
      |              ^~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:918:20: note: in expansion of macro 'StaticAssertStmt'
  918 |         ((void) ({ StaticAssertStmt(condition, errmessage); true; }))
      |                    ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:962:10: note: in expansion of macro 'StaticAssertExpr'
  962 |         (StaticAssertExpr(__builtin_types_compatible_p(__typeof__(varname), typename), \
      |          ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/postgres.h:550:10: note: in expansion of macro 'AssertVariableIsOfTypeMacro'
  550 |         (AssertVariableIsOfTypeMacro(X, int64), Int64GetDatum(X))
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:825:39: note: in expansion of macro 'Int64GetDatumFast'
  825 |       values[MEMORY_INFO_SWAP_FREE] = Int64GetDatumFast(swap_free_bytes);
      |                                       ^~~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:916:14: error: static assertion failed: "cached_bytes does not have type int64"
  916 |         do { _Static_assert(condition, errmessage); } while(0)
      |              ^~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:918:20: note: in expansion of macro 'StaticAssertStmt'
  918 |         ((void) ({ StaticAssertStmt(condition, errmessage); true; }))
      |                    ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:962:10: note: in expansion of macro 'StaticAssertExpr'
  962 |         (StaticAssertExpr(__builtin_types_compatible_p(__typeof__(varname), typename), \
      |          ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/postgres.h:550:10: note: in expansion of macro 'AssertVariableIsOfTypeMacro'
  550 |         (AssertVariableIsOfTypeMacro(X, int64), Int64GetDatum(X))
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:826:41: note: in expansion of macro 'Int64GetDatumFast'
  826 |       values[MEMORY_INFO_CACHE_TOTAL] = Int64GetDatumFast(cached_bytes);
      |                                         ^~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c: In function 'network_info':
/usr/pgsql-16/include/server/c.h:916:14: error: static assertion failed: "tx_bytes does not have type int64"
  916 |         do { _Static_assert(condition, errmessage); } while(0)
      |              ^~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:918:20: note: in expansion of macro 'StaticAssertStmt'
  918 |         ((void) ({ StaticAssertStmt(condition, errmessage); true; }))
      |                    ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:962:10: note: in expansion of macro 'StaticAssertExpr'
  962 |         (StaticAssertExpr(__builtin_types_compatible_p(__typeof__(varname), typename), \
      |          ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/postgres.h:550:10: note: in expansion of macro 'AssertVariableIsOfTypeMacro'
  550 |         (AssertVariableIsOfTypeMacro(X, int64), Int64GetDatum(X))
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:960:42: note: in expansion of macro 'Int64GetDatumFast'
  960 |          values[NETWORK_INFO_TX_BYTES] = Int64GetDatumFast(tx_bytes);
      |                                          ^~~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:916:14: error: static assertion failed: "tx_packets does not have type int64"
  916 |         do { _Static_assert(condition, errmessage); } while(0)
      |              ^~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:918:20: note: in expansion of macro 'StaticAssertStmt'
  918 |         ((void) ({ StaticAssertStmt(condition, errmessage); true; }))
      |                    ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:962:10: note: in expansion of macro 'StaticAssertExpr'
  962 |         (StaticAssertExpr(__builtin_types_compatible_p(__typeof__(varname), typename), \
      |          ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/postgres.h:550:10: note: in expansion of macro 'AssertVariableIsOfTypeMacro'
  550 |         (AssertVariableIsOfTypeMacro(X, int64), Int64GetDatum(X))
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:961:44: note: in expansion of macro 'Int64GetDatumFast'
  961 |          values[NETWORK_INFO_TX_PACKETS] = Int64GetDatumFast(tx_packets);
      |                                            ^~~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:916:14: error: static assertion failed: "tx_errors does not have type int64"
  916 |         do { _Static_assert(condition, errmessage); } while(0)
      |              ^~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:918:20: note: in expansion of macro 'StaticAssertStmt'
  918 |         ((void) ({ StaticAssertStmt(condition, errmessage); true; }))
      |                    ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:962:10: note: in expansion of macro 'StaticAssertExpr'
  962 |         (StaticAssertExpr(__builtin_types_compatible_p(__typeof__(varname), typename), \
      |          ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/postgres.h:550:10: note: in expansion of macro 'AssertVariableIsOfTypeMacro'
  550 |         (AssertVariableIsOfTypeMacro(X, int64), Int64GetDatum(X))
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:962:43: note: in expansion of macro 'Int64GetDatumFast'
  962 |          values[NETWORK_INFO_TX_ERRORS] = Int64GetDatumFast(tx_errors);
      |                                           ^~~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:916:14: error: static assertion failed: "tx_dropped does not have type int64"
  916 |         do { _Static_assert(condition, errmessage); } while(0)
      |              ^~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:918:20: note: in expansion of macro 'StaticAssertStmt'
  918 |         ((void) ({ StaticAssertStmt(condition, errmessage); true; }))
      |                    ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:962:10: note: in expansion of macro 'StaticAssertExpr'
  962 |         (StaticAssertExpr(__builtin_types_compatible_p(__typeof__(varname), typename), \
      |          ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/postgres.h:550:10: note: in expansion of macro 'AssertVariableIsOfTypeMacro'
  550 |         (AssertVariableIsOfTypeMacro(X, int64), Int64GetDatum(X))
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:963:44: note: in expansion of macro 'Int64GetDatumFast'
  963 |          values[NETWORK_INFO_TX_DROPPED] = Int64GetDatumFast(tx_dropped);
      |                                            ^~~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:916:14: error: static assertion failed: "rx_bytes does not have type int64"
  916 |         do { _Static_assert(condition, errmessage); } while(0)
      |              ^~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:918:20: note: in expansion of macro 'StaticAssertStmt'
  918 |         ((void) ({ StaticAssertStmt(condition, errmessage); true; }))
      |                    ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:962:10: note: in expansion of macro 'StaticAssertExpr'
  962 |         (StaticAssertExpr(__builtin_types_compatible_p(__typeof__(varname), typename), \
      |          ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/postgres.h:550:10: note: in expansion of macro 'AssertVariableIsOfTypeMacro'
  550 |         (AssertVariableIsOfTypeMacro(X, int64), Int64GetDatum(X))
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:964:42: note: in expansion of macro 'Int64GetDatumFast'
  964 |          values[NETWORK_INFO_RX_BYTES] = Int64GetDatumFast(rx_bytes);
      |                                          ^~~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:916:14: error: static assertion failed: "rx_packets does not have type int64"
  916 |         do { _Static_assert(condition, errmessage); } while(0)
      |              ^~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:918:20: note: in expansion of macro 'StaticAssertStmt'
  918 |         ((void) ({ StaticAssertStmt(condition, errmessage); true; }))
      |                    ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:962:10: note: in expansion of macro 'StaticAssertExpr'
  962 |         (StaticAssertExpr(__builtin_types_compatible_p(__typeof__(varname), typename), \
      |          ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/postgres.h:550:10: note: in expansion of macro 'AssertVariableIsOfTypeMacro'
  550 |         (AssertVariableIsOfTypeMacro(X, int64), Int64GetDatum(X))
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:965:44: note: in expansion of macro 'Int64GetDatumFast'
  965 |          values[NETWORK_INFO_RX_PACKETS] = Int64GetDatumFast(rx_packets);
      |                                            ^~~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:916:14: error: static assertion failed: "rx_errors does not have type int64"
  916 |         do { _Static_assert(condition, errmessage); } while(0)
      |              ^~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:918:20: note: in expansion of macro 'StaticAssertStmt'
  918 |         ((void) ({ StaticAssertStmt(condition, errmessage); true; }))
      |                    ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:962:10: note: in expansion of macro 'StaticAssertExpr'
  962 |         (StaticAssertExpr(__builtin_types_compatible_p(__typeof__(varname), typename), \
      |          ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/postgres.h:550:10: note: in expansion of macro 'AssertVariableIsOfTypeMacro'
  550 |         (AssertVariableIsOfTypeMacro(X, int64), Int64GetDatum(X))
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:966:43: note: in expansion of macro 'Int64GetDatumFast'
  966 |          values[NETWORK_INFO_RX_ERRORS] = Int64GetDatumFast(rx_errors);
      |                                           ^~~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:916:14: error: static assertion failed: "rx_dropped does not have type int64"
  916 |         do { _Static_assert(condition, errmessage); } while(0)
      |              ^~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:918:20: note: in expansion of macro 'StaticAssertStmt'
  918 |         ((void) ({ StaticAssertStmt(condition, errmessage); true; }))
      |                    ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:962:10: note: in expansion of macro 'StaticAssertExpr'
  962 |         (StaticAssertExpr(__builtin_types_compatible_p(__typeof__(varname), typename), \
      |          ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/postgres.h:550:10: note: in expansion of macro 'AssertVariableIsOfTypeMacro'
  550 |         (AssertVariableIsOfTypeMacro(X, int64), Int64GetDatum(X))
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:967:44: note: in expansion of macro 'Int64GetDatumFast'
  967 |          values[NETWORK_INFO_RX_DROPPED] = Int64GetDatumFast(rx_dropped);
      |                                            ^~~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:916:14: error: static assertion failed: "speed_mbps does not have type int64"
  916 |         do { _Static_assert(condition, errmessage); } while(0)
      |              ^~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:918:20: note: in expansion of macro 'StaticAssertStmt'
  918 |         ((void) ({ StaticAssertStmt(condition, errmessage); true; }))
      |                    ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/c.h:962:10: note: in expansion of macro 'StaticAssertExpr'
  962 |         (StaticAssertExpr(__builtin_types_compatible_p(__typeof__(varname), typename), \
      |          ^~~~~~~~~~~~~~~~
/usr/pgsql-16/include/server/postgres.h:550:10: note: in expansion of macro 'AssertVariableIsOfTypeMacro'
  550 |         (AssertVariableIsOfTypeMacro(X, int64), Int64GetDatum(X))
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
rpm/redhat/main/non-common/pgexporter_ext/main/pgexporter_ext-0.2.2/src/pgexporter_ext/lib.c:968:44: note: in expansion of macro 'Int64GetDatumFast'
  968 |          values[NETWORK_INFO_LINK_SPEED] = Int64GetDatumFast(speed_mbps);
      |                                            ^~~~~~~~~~~~~~~~~

**Comment by jesperpedersen** [comment]
What is your `gcc --version` ?

**Comment by jesperpedersen** [comment]
@resyfer Ideas ?

**Comment by devrimgunduz** [comment]
> What is your `gcc --version` ?

13.2 - Fedora 38.

**Comment by resyfer** [comment]
@jesperpedersen I have reproduced the compile error. I'll be looking into it. The error suggests a type problem, will have to look into it a bit.

**Comment by resyfer** [comment]
@jesperpedersen I made this change:

```diff
Datum
pgexporter_information_ext(PG_FUNCTION_ARGS)
{
   Datum version;
   ...
-  PG_RETURN_CSTRING(version);
+  PG_RETURN_DATUM(version);
}
```

This was the cause of the error...This builds `pgexporter_ext` without issues for both 15 & 16. There were some other minor `uint64_t` & `int64_t` mismatch as well.

I've added #12 for all of it. I am not entirely sure how it was building without issues with PostgreSQL 15... would have to probably look into that changes related to it to solve the mystery I think.
